extensions/kotlin-dsl/deployment/src/main/java/org/apache/camel/quarkus/dsl/kotlin/deployment/KotlinDslProcessor.java

```diff
@@ -0,0 +1,224 @@
+import java.util.stream.Stream;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedClassBuildItem;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.dev.CompilationProvider;
+import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeBuild;
+import io.quarkus.kotlin.deployment.KotlinCompilationProvider;
+import io.quarkus.maven.dependency.ResolvedDependency;
+import io.quarkus.paths.PathCollection;
+import io.quarkus.runtime.RuntimeValue;
+import org.apache.camel.CamelContext;
+import org.apache.camel.dsl.kotlin.KotlinConstantsKt;
+import org.apache.camel.quarkus.core.deployment.main.CamelMainHelper;
+import org.apache.camel.quarkus.core.deployment.spi.CamelContextBuildItem;
+import org.apache.camel.quarkus.dsl.kotlin.runtime.Configurer;
+import org.apache.camel.quarkus.dsl.kotlin.runtime.KotlinDslRecorder;
+import org.apache.camel.spi.Resource;
+import org.apache.camel.util.FileUtil;
+import org.apache.camel.util.IOHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class KotlinDslProcessor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KotlinDslProcessor.class);
+    private static final String PACKAGE_NAME = "org.apache.camel.quarkus.dsl.kotlin.generated";
+    private static final Pattern IMPORT_PATTERN = Pattern.compile("import .*");
+    private static final String FILE_FORMAT = "package %s\n" +
+            "%s\n" +
+            "class %s(builder: org.apache.camel.builder.endpoint.EndpointRouteBuilder) :\n" +
+            "  %s(builder) { \n" +
+            "  override fun configure() { \n" +
+            "    %s\n" +
+            "  }\n" +
+            "}";
+    private static final String FEATURE = "camel-kotlin-dsl";
+    public static final String CLASS_EXT = ".class";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep(onlyIf = NativeBuild.class)
+    void compileScriptsAOT(BuildProducer<GeneratedClassBuildItem> generatedClass,
+            BuildProducer<GeneratedResourceBuildItem> generatedResource,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            BuildProducer<KotlinGeneratedClassBuildItem> generatedKotlinClass,
+            CurateOutcomeBuildItem curateOutcomeBuildItem) throws Exception {
+        LOG.debug("Loading .kts resources");
+        final Path projectDir = Paths.get(".").toAbsolutePath().normalize();
+        final Path generatedSourceDir = projectDir.resolve("target/kotlin-dsl/generated-sources");
```

Comment:
This might fail on Gradle or even with some non-standard Maven setup. 
`io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem` or something similar provided by Quarkus Core could perhaps deliver safer results.

Reply:
Nice one, thx for sharing